tiling.py

```diff
@@ -25,18 +28,43 @@ def __init__(self, height, up, mirror1_ctr, mirror2_ctr):
             shapes.Plane(ctr1, ctr1 + up_proj, ctr1 + right_proj),
             shapes.Plane(ctr2, ctr2 + up_proj, ctr2 + right_proj)
         ]
+
+        #add floor mirror plane if we only have floor mirror else add ceiling mirror plane
+        if not self._flag_ceiling and self._flag_floor:
+            self._vmirror_planes = shapes.Plane((0, 0, 1), (0, 0, 0), (1, 0, 0))
+        else:
+            self._vmirror_planes = shapes.Plane((0, height, 1), (0, height, 0), (1, height, 0))
+
```

Comment:
I'm pretty sure line 32 is a bug (the not is attached to the first clause not the whole thing)

Also you didn't transform the plane into the reference basis so this will be a bug if your scene isn't aligned with the standard basis.

Did you test this?

I would rewrite this as:
```
self._ground_plane = shapes.Plane(b, b + m1n, b + right_proj) # Move the last line of the constructor
self._ceiling_plane = self._ground_plane.transform(translate(height*up))
```

No if statements, and you don't need to add this vmirrorplanes variable.